research/picai/nnunet_scripts/nnunet_launch.slrm

```diff
@@ -0,0 +1,81 @@
+#!/bin/bash
+
+#SBATCH --gres=gpu:a40:1
+#SBATCH --qos=normal
+#SBATCH --mem=64GB
+#SBATCH -c 16
+#SBATCH --time=12:00:00
+#SBATCH --signal=B:USR1@60
+#SBATCH --job-name=nnunet_picai_train
+#SBATCH --output=%j_%x.out
+#SBATCH --error=%j_%x.err
+
+# Process Inputs
+DATASET_NAME=$1
+UNET_CONFIG=$2
+FOLD=$3
+VENV_PATH=$4
+PLANS_IDENTIFIER=${5:-'plans'} # Default value of plans string
+PRETAINED_WEIGHTS=${6:-''} # Default value of empty string
+
+# Log input information
+echo "Dataset Name : ${DATASET_NAME}"
+echo "Config: ${UNET_CONFIG}"
+echo "Dataset Fold: ${FOLD}"
+echo "Python Env Path: ${VENV_PATH}"
+echo "Plans Identifier: ${PLANS_IDENTIFIER}"
+
+if [[ -z "${PRETRAINED_WEIGHTS}" ]]; then
+	echo "Pretrained Weights: ${PRETRAINED_WEIGHTS}"
+fi
+
+# Setup before running training
+source ${VENV_PATH}bin/activate
+BASE_DIR="$nnUNet_results/${DATASET_NAME}/nnUNetTrainer__${PLANS_IDENTIFIER}__${UNET_CONFIG}/"
+
+# Function returning array of args passed to nnUNetv2_train command
+get_nnunet_train_args() {
+	local checkpoint_bool=$1
+	local arg_list = ( $DATASET_NAME $UNET_CONFIG $FOLD $PLANS_IDENTIFIER)
+
+	if [[ -z "${PRETRAINED_WEIGHTS}" ]]; then
+		arg_list += ${PRETRAINED_WEIGHTS}
+	fi
+	if [ "$checkpoint_bool" = true ]; then
+		arg_list += "--c"
+	fi
+	return args_list
+}
+
+# Function that is called to requeue current slurm job
+handler()
+{
+	echo "Requeue $SLURM_JOB_ID at $(date)"
+	scontrol requeue $SLURM_JOB_ID
+}
+
+# Function that checks whether to start or resume training, gets train_args
+# and runs  training
+nnunet_train()
+{
+	if [ -z "$(ls -A $BASE_DIR)" ]; then
+		echo "Start training from scratch as $(date)"
+		get_nnunet_train_args false
+		local train_args = $?
+	else
+		echo "Start training from checkpoint as $(date)"
+		get_nnunet_train_args true
```

Comment:
Interesting, I've never seen this before. Maybe this is the standard bash workflow for calling a function and capturing its values?